frontend/packages/pipelines-plugin/src/components/quicksearch/__tests__/PipelineQuicksearchDetails.spec.tsx

```diff
@@ -0,0 +1,146 @@
+    closeModal: jest.fn(),
+  };
+
+  afterEach(() => cleanup());
+
+  describe('Installed badge tests', () => {
+    it('should show the installed badge for the cluster task', async () => {
+      const { queryByTestId } = render(<PipelineQuickSearchDetails {...cluterTaskProps} />);
+      await waitFor(() => {
+        expect(queryByTestId('task-installed-badge')).not.toBeNull();
+      });
+    });
+
+    test('should show the installed badge for the installed tekton hub task', async () => {
+      const installedTektonHubTask = {
+        ...sampleTektonHubCatalogItem,
+        attributes: { ...sampleTektonHubCatalogItem.attributes, installed: 1 },
+      };
+      const { queryByTestId } = render(
+        <PipelineQuickSearchDetails
+          {...{ ...tektonHubProps, selectedItem: installedTektonHubTask }}
+        />,
+      );
+      await waitFor(() => {
+        expect(queryByTestId('task-installed-badge')).not.toBeNull();
+      });
+    });
+
+    test('should not show the installed badge for the uninstalled tekton hub task', async () => {
+      const { queryByTestId } = render(<PipelineQuickSearchDetails {...tektonHubProps} />);
+      await waitFor(() => {
+        expect(queryByTestId('task-installed-badge')).toBeNull();
+      });
+    });
+  });
+
+  describe('CTA button tests', () => {
+    test('should show the Add button for already installed task', async () => {
+      const { getByRole } = render(<PipelineQuickSearchDetails {...cluterTaskProps} />);
+      await waitFor(() => {
+        expect(getByRole('button', { name: 'Add' })).not.toBeNull();
+      });
+    });
+
+    test('should show the Install and add button for uninstalled tekton hub task', async () => {
+      const { getByRole } = render(<PipelineQuickSearchDetails {...tektonHubProps} />);
+      await waitFor(() => {
+        expect(getByRole('button', { name: 'Install and add' })).not.toBeNull();
+      });
+    });
+
+    test('should show the Update and add button for already installed task', async () => {
+      const installedTektonHubTask = {
+        ...sampleTektonHubCatalogItem,
+        attributes: { ...sampleTektonHubCatalogItem.attributes, installed: 1 },
+      };
+      const { getByRole, queryByTestId } = render(
+        <PipelineQuickSearchDetails
+          {...{ ...tektonHubProps, selectedItem: installedTektonHubTask }}
+        />,
+      );
+      await waitFor(async () => {
+        fireEvent.click(queryByTestId('task-version-toggle'));
+        fireEvent.click(screen.getByText('0.2'));
+        expect(getByRole('button', { name: 'Update and add' })).not.toBeNull();
+      });
+    });
+  });
+
+  describe('Version dropdown tests', () => {
+    test('should show the version dropdown if the versions are available', async () => {
+      const { queryByTestId } = render(<PipelineQuickSearchDetails {...cluterTaskProps} />);
+      await waitFor(() => {
+        expect(queryByTestId('task-version')).not.toBeNull();
```

Comment:
Hmmm... I wonder about digging into other components to get their test ids... seems a little odd. Can we search for the component name instead?

Reply:
Removed the querying child components here.